### ISSUE TITLE
[research] Diffusion eval metric (masked-CE / ELBO bound)

### ISSUE THREAD

**Issue by travisgalloway** [travisgalloway/monica#46]
Part of #42. Depends on #D2.

**Goal:** A held-out metric appropriate to masked diffusion, since the autoregressive `val_perplexity` in `src/eval/val_loss.py` is not the right yardstick (diffusion doesn't predict next-token left-to-right). Report the masked-token cross-entropy and an ELBO bound (the `1/t`-weighted masked CE integrated over the noise schedule), optionally generative perplexity from the sampler.

**Acceptance criteria**
- [ ] New portable `src/eval/diffusion_loss.py` (no `mlx`/`torch`), added to `PORTABLE_MODULES`, taking a `to_numpy` converter at the seam (mirroring `src/eval/val_loss.py`).
- [ ] Computes masked-CE and the ELBO bound over a held-out split; numerically stable softmax (reuse the `val_loss.py` pattern).
- [ ] Unit test on synthetic logits/targets with a known answer.

**Files:** `src/eval/diffusion_loss.py` (new, portable), `tests/test_import_guard.py`, `tests/test_diffusion_loss.py` (new).

Seam note: portable; consumes numpy logits via a `to_numpy` callable like `val_loss.py`.

## Verification
1. `.venv/bin/python -m pytest tests/test_diffusion_loss.py tests/test_import_guard.py`

**Comment by travisgalloway** [comment]
Consolidated into and owned by #203 ('[M12] P4-P5: diffusion arm + LSP discriminator'), which absorbs the discrete-diffusion research track. Closed during closure audit to groom the backlog.
